staging/src/k8s.io/pod-security-admission/test/fixtures_hostPorts.go

```diff
@@ -0,0 +1,164 @@
+			p = ensureHostPorts(p)
+			return []*corev1.Pod{
+				p,
+				// no host ports
+				tweak(p, func(p *corev1.Pod) {
+					p.Spec.Containers[0].Ports = []corev1.ContainerPort{
+						corev1.ContainerPort{
+							ContainerPort: 12345,
+						},
+					}
+				}),
+				tweak(p, func(p *corev1.Pod) {
+					p.Spec.InitContainers[0].Ports = []corev1.ContainerPort{
+						corev1.ContainerPort{
+							ContainerPort: 12346,
+						},
+					}
+				}),
+			}
+		},
+		generateFail: func(p *corev1.Pod) []*corev1.Pod {
+			p = ensureHostPorts(p)
+			return []*corev1.Pod{
+				// Host Port present
+				tweak(p, func(p *corev1.Pod) {
+					p.Spec.Containers[0].Ports = []corev1.ContainerPort{
+						corev1.ContainerPort{
+							ContainerPort: 12345,
+							HostPort:      12345,
+						},
+					}
+				}),
+				tweak(p, func(p *corev1.Pod) {
+					p.Spec.InitContainers[0].Ports = []corev1.ContainerPort{
+						corev1.ContainerPort{
+							ContainerPort: 12346,
+							HostPort:      12346,
+						},
+					}
+				}),
+				// both init-container and app container use host port
+				tweak(p, func(p *corev1.Pod) {
+					p.Spec.Containers[0].Ports = []corev1.ContainerPort{
+						corev1.ContainerPort{
+							ContainerPort: 12345,
+							HostPort:      12345,
+						},
+					}
+					p.Spec.InitContainers[0].Ports = []corev1.ContainerPort{
+						corev1.ContainerPort{
+							ContainerPort: 12346,
+							HostPort:      12346,
+						},
+					}
+				}),
+				// no ports for init-container and app container uses host port
+				tweak(p, func(p *corev1.Pod) {
+					p.Spec.Containers[0].Ports = []corev1.ContainerPort{
+						corev1.ContainerPort{
+							ContainerPort: 12345,
+							HostPort:      12345,
+						},
+					}
+					p.Spec.InitContainers[0].Ports = nil
+				}),
+
+				// no host port for init container and app container uses host port
+				tweak(p, func(p *corev1.Pod) {
+					p.Spec.Containers[0].Ports = []corev1.ContainerPort{
+						corev1.ContainerPort{
+							ContainerPort: 12345,
+							HostPort:      12345,
+						},
+					}
+					p.Spec.InitContainers[0].Ports = []corev1.ContainerPort{
+						corev1.ContainerPort{
+							ContainerPort: 12346,
+						},
+					}
+				}),
+
+				// no ports for app container and init container uses host port
+				tweak(p, func(p *corev1.Pod) {
+					p.Spec.Containers[0].Ports = nil
+					p.Spec.InitContainers[0].Ports = []corev1.ContainerPort{
+						corev1.ContainerPort{
+							ContainerPort: 12346,
+							HostPort:      12346,
+						},
+					}
+				}),
+				// no host port for app container and init container uses host port
+				tweak(p, func(p *corev1.Pod) {
+					p.Spec.Containers[0].Ports = []corev1.ContainerPort{
+						corev1.ContainerPort{
+							ContainerPort: 12345,
+						},
+					}
+					p.Spec.InitContainers[0].Ports = []corev1.ContainerPort{
+						corev1.ContainerPort{
+							ContainerPort: 12346,
+							HostPort:      12346,
+						},
+					}
+				}),
```

Comment:
The first two failure scenarios make sure we're specifically checking both containers and initcontainers; I'd suggest just one additional failure scenario that populates both container and hostports to make sure we properly detect hostports in a mix of nonhomogenous ports

That would give us three failure scenarios that provides a good amount of coverage

Reply:
Oh ok. Was going by this description in the issue verbatim:

> - mix of hostPorts and non-hostPorts in containers and initContainers

Are you suggesting keeping only these failure scenarios and remove others?
1. app uses host port. Init container does not use any ports
2. init container uses host port. App container does not use any ports
3. app container uses host port. Init container uses host port

Reply:
yes to 1 and 2

for 3, put regular ports and hostPorts in both containers and initContainers

Reply:
Ok, Fixed it in my last commit.